src/io/ResultSet.jl

```diff
@@ -15,15 +15,15 @@ const ENV_STATS = "env_stats"
 const MODEL_SPEC = "model_spec"
 
 
-struct ResultSet{S,T1,T2,F,A,B,C,D,G,D1,D2,D3,DF}
-    name::S
-    RCP::S
-    invoke_time::S
-    ADRIA_VERSION::S
+struct ResultSet{T1,T2,A,B,C,D,G,D1,D2,D3,DF}
+    name::String
+    RCP::String
+    invoke_time::String
+    ADRIA_VERSION::String
```

Comment:
I'm curious about this change. Is this to improve performance in some way?

Reply:
It's a minor change, probably won't do much but I think it would reduce initial compile times a little.

Each field "type" specified like so:

`ResultSet{T1,T2,A,B,C,D,G,D1,D2,D3,DF}`

Effectively resolves to `Any` because no other information is given. So the Julia compiler does not create any optimised code at compile time - this is deferred to runtime instead. At the same time, it means the Julia compiler has to be ready to compile combinations of any type. 

Specifying the known types directly helps reduce the number of possible combinations.

It's currently set up like this because I wasn't sure exactly what the structure of `ResultSet` would be, and it's likely to continue changing as our needs/use context evolve.

See: 
1. https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-type
2. https://docs.julialang.org/en/v1/manual/performance-tips/#Avoid-fields-with-abstract-containers